qiita_db/metadata_template/test/test_sample_template.py

```diff
@@ -1343,10 +1343,13 @@ def test_update_numpy(self):
 
         sql = "SELECT * FROM qiita.sample_{0}".format(st.id)
         obs = self.conn_handler.execute_fetchall(sql)
-        exp = [['%s.Sample1' % self.new_study.id, 'false',
-                '2015-09-01 00:00:00'],
-               ['%s.Sample2' % self.new_study.id, 'true',
-                '2015-09-01 00:00:00']]
+        exp = [
+            ['%s.Sample2' % self.new_study.id, {
+                'bool_col': 'true', 'date_col': '2015-09-01 00:00:00'}],
```

Comment:
We want to be careful here:

```python
In [11]: json.loads('{"foo": true}')
Out[11]: {'foo': True}

In [12]: json.loads('{"foo": "true"}')
Out[12]: {'foo': 'true'}
```

Reply:
Good point! For our peace of mind, we always load the info via pandas as str (dtype=str) and use the df.to_json

Reply:
There was a lot of use of `json.dumps` which is going to likely have a similar issue. I don't see how the use of pandas protects us from implicit behavior of the json encoders / decoders

Reply:
pandas dtype=str load (or should load) everything as string; thus, pandas will not try to infer datatypes. From my own experience, tools will use the dtype of pandas to infer the datatype of the column. Thus, if in pandas everything is str, the other tools will use as the type.

```
In [1]: import pandas as pd

In [2]: pd.DataFrame.from_dict({'val': {'row': 5}}).dtypes
Out[2]: 
val    int64
dtype: object

In [3]: pd.DataFrame.from_dict({'val': {'row': 5}}, dtype=str).dtypes
Out[3]: 
val    object
dtype: object
```